citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java

```diff
@@ -514,8 +514,10 @@ private void displayCitations(PrintWriter out, List<Citation> citations, Citatio
 				}
 			} else {
 				// We only want to show the open url if no custom urls have been specified.
-				out.println("\t\t\t\t<a href=\"" + ((Vector)citation.getCitationProperty("otherIds")).get(0) + "\" target=\"_blank\">"
-						+ "Find it" + " on SOLO" + "</a>");
+				if (citation.getCitationProperty("otherIds") instanceof Vector) {
+					out.println("\t\t\t\t<a href=\"" + ((Vector) citation.getCitationProperty("otherIds")).get(0) + "\" target=\"_blank\">"
```

Comment:
I think you can just change it to `citation.getCitationProperty("otherIds", true)` which will work in all cases (single and multivalued).